### DISCUSSION TITLE
video stim

### DISCUSSION THREAD

**Discussion by jwitvoet** [jspsych/jsPsych#265] [Q&A]
I'm having trouble with using video's as stimuli.

In my experiment I want to ask participants if they think it's save to cross a road. Participants will watch a video and will be asked to judge the gaps between oncoming traffic as either save or unsafe to cross (by pressing two keys while watching the video). 

From what i've read so far in other posts, a code similar to this should suffice:

var trial = {
  type: 'single-stim',
  stimulus: '<video autoplay><source src="myvideo.mp4" type="video/mp4"></video>',
  prompt: 'Press Y if the video has a gorilla in it, press N otherwise.',
  choices: ['y','n'],
  is_html: true
}

However, when I do this the video does not play. The video does play in a html file, but not with jsPsych. 

In the html file I use the following code:

<video controls autoplay preload width="720" height="460">
    <source src="video1.mp4" type="video/mp4">
    <source src="video1.ogv" type="video/ogv">
      <source src="video1.webm" type="video/webm">
    <object type="application/x-shockwave-flash" data="http://player.longtailvideo.com/player.swf" width="640" height="360">
        <param name="movie" value="http://player.longtailvideo.com/player.swf">
        <param name="allowFullScreen" value="true">
        <param name="wmode" value="transparent">
        <param name="flashVars" value="autostart=true&amp;controlbar=over&amp;file=https%3A%2F%2Fyoutu.be%2Fgm00-FdjWns">
        <span title="No video playback capabilities, please download the video below">test</span>
    </object>
</video>

I'm a newbie at using jsPsych, so some insight in to what i'm doing wrong would be much appreciated.

**Comment by jodeleeuw** [comment]
It doesn't look like the code above has any reference to the video file in it. Did you mean to paste a different set of code?

**Comment by jwitvoet** [comment]
Thanks for the quick reply. See below for the code. The video is stored locally on my computer in the same file as the experiment. I hope you can spot the problem.  

<!doctype html>
<html>
    <head>
        <title>Onderdeel 4</title>
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
        <script src="jspsych-5.0.3/jspsych.js"></script>
        <script src="jspsych-5.0.3/plugins/jspsych-text.js"></script>
        <script src="jspsych-5.0.3/plugins/jspsych-single-stim.js"></script>
        <script src="jspsych-5.0.3/plugins/jspsych-instructions.js"></script>
        <link href="jspsych-5.0.3/css/jspsych.css" rel="stylesheet" type="text/css"></link>
    </head>
    <body>
    </body>
    <script>

    var welcome_block = {
        type: 'instructions',
        pages: ['Dit is onderdeel 4.'],
        show_clickable_nav:true
    };

    var uitleg = {
        type: 'instructions',
        pages: ['Stel je voor dat je op de fiets een weg over wilt steken.'],
        show_clickable_nav:true
    };

   var trial = {
  type: 'single-stim',
  stimulus: '<video preload autoplay controls>'+
    <source src="video1.mp4" type="video/mp4">+
    <source src="video1.ogv" type="video/ogv">+
    <source src="video1.webm" type="video/webm">+
  </video>',
  prompt: 'Press Y if safe, press N if unsafe.',
  choices: ['y','n'],
  is_html: true
}

    var timeline = [];
timeline.push(welcome_block);
timeline.push(uitleg);
timeline.push(trial)


    jsPsych.init({
        timeline: timeline
    })

    </script>
</html>

**Comment by jwitvoet** [comment]
apparently my source code is removed when I post it. I've attached it in a file. 

[experiment.pdf](https://github.com/jodeleeuw/jsPsych/files/314283/experiment.pdf)

**Comment by jodeleeuw** [comment]
It may be related to local file:// protocol issues. See [this stackoverflow thread](http://stackoverflow.com/questions/8885701/play-local-hard-drive-video-file-with-html5-video-tag).

**Comment by jwitvoet** [comment]
I figured it out. Thank you for your help.
